File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.java
class Solution {
    public int kthSmallest(int[][] mat, int k) {
        //if(mat.length==1 && k==1) return mat[0][0];
         PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        
        // Add all elements to the heap
        for (int i = 0; i < mat.length; i++) {
            for (int j = 0; j < mat[0].length; j++) {
                minHeap.offer(mat[i][j]);
            }
        }
        
        // Extract the first k-1 elements from the heap
        for (int i = 1; i < k; i++) {
            minHeap.poll();
        }
        
        // The kth smallest element is now at the top of the heap
        return minHeap.poll();
    }
}

File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.py
class Solution(object):
    def kthSmallest(self, matrix, k):
        """
        :type matrix: List[List[int]]
        :type k: int
        :rtype: int
        """
        heap=[]
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                heapq.heappush(heap,matrix[i][j])
        while k>1:
            heapq.heappop(heap)
            k-=1
        return heapq.heappop(heap)
        